subscription/views.py

```diff
@@ -1016,39 +1026,75 @@ def post(self, request):
+    for pattern_entry in UriPattern.objects.all():
+        try:
+            # Compile the pattern from the UriPattern table
+            regex_pattern = re.compile(pattern_entry.pattern)
+            # Match the full URL against the compiled pattern
+            if regex_pattern.search(url):
+                uri_pattern_object = pattern_entry
+                logger.debug("Match found for pattern: %s" % pattern_entry.pattern)
+                break
+        except re.error as e:
+            logger.error("Regex compilation failed for pattern: %s with error: %s" % (pattern_entry.pattern, e))
+
+    # If no match is found in UriPattern table, return 0
+    if not uri_pattern_object:
+        logger.debug("No matching pattern found in UriPattern table.")
+        return FREE_ACCESS_UNIT
+
+    # Step 2: Look up usage units from PremiumUsageUnits using the found UriPattern entry
     try:
-        # First, try an exact match
-        premium_page = PremiumUsageUnits.objects.get(url=url)
+        # Use the foreign key relationship to look up units_consumed in PremiumUsageUnits
+        premium_page = PremiumUsageUnits.objects.get(pattern_object=uri_pattern_object)
+        logger.debug("Units found for matched pattern: %d" % premium_page.units_consumed)
         return premium_page.units_consumed
     except PremiumUsageUnits.DoesNotExist:
-        # If no exact match, try containment check
-        for premium_page in PremiumUsageUnits.objects.all():
-            if premium_page.url in url:
-                return premium_page.units_consumed
-        
-        # If still no match, try regex matching as a fallback
-        for premium_page in PremiumUsageUnits.objects.all():
-            if re.match(premium_page.url, url):
-                return premium_page.units_consumed
-    
-    return 1  # Default to 1 unit if not found in PremiumUsageUnits
+        # If no specific PremiumUsageUnits entry is found, return default value of 1
+        logger.debug("No PremiumUsageUnits entry found for pattern id: %d. Defaulting to 1 unit." % uri_pattern_object.patternId)
+        return PAID_ACCESS_UNIT
+
 
 class CheckLimit(APIView):
     requireApiKey = False
 
     def get(self, request):
         complete_uri = request.GET.get('uri')
         party_id = request.GET.get('party_id')
+        partnerId = request.GET.get('partner_id')
 
-        if not all([party_id, complete_uri]):
+        if not all([party_id, complete_uri, partnerId]):
             return Response({"error": "Missing required parameters"}, status=status.HTTP_400_BAD_REQUEST)
 
-        units_required = get_premium_units(complete_uri)
-        
+        units_required = get_usage_units(complete_uri)
+        logger.debug("units_required: " + str(units_required))
         try:
-            limit_value_object = LimitValue.objects.get(limitId=1)
+            limit_value_object = LimitValue.objects.get(partnerId=partnerId)
```

Comment:
You may get a "MultipleObjectsReturned" error if the db has multiple values, need to handle it as well (you cannot assume/guarantee that the DB only has one value)